cmplibrary/src/test/java/com/sourcepoint/gdpr_cmplibrary/GDPRUserConsentTest.java

```diff
@@ -0,0 +1,182 @@
+                "\"IABTCF_PurposeConsents\":\"111111111100000000";
+
+    private GDPRUserConsent userConsent;
+    private JSONObject jsonConsentsMock;
+    private String consentUUIDMock;
+
+    @Before
+    public void setUp() throws JSONException, ConsentLibException {
+        JSONObject consentsResponse = new JSONObject(jsonData);
+        consentUUIDMock = consentsResponse.getString("uuid");
+        jsonConsentsMock = consentsResponse.getJSONObject("userConsent");
+        userConsent = new GDPRUserConsent(jsonConsentsMock ,consentUUIDMock);
+    }
+
+    @Test
+    public void emptyUserConsentConstructor() {
+        GDPRUserConsent emptyUserConsent = new GDPRUserConsent();
+
+        assertEquals(DEFAULT_EMPTY_UUID, emptyUserConsent.uuid);
+        assertEquals(DEFAULT_EMPTY_CONSENT_STRING, emptyUserConsent.consentString);
+        assertTrue(emptyUserConsent.acceptedCategories.isEmpty());
+        assertTrue(emptyUserConsent.acceptedVendors.isEmpty());
+        assertTrue(emptyUserConsent.legIntCategories.isEmpty());
+        assertTrue(emptyUserConsent.specialFeatures.isEmpty());
+        assertTrue(emptyUserConsent.TCData.isEmpty());
+        assertTrue(emptyUserConsent.vendorGrants.isEmpty());
+    }
+
+
+    @Test
+    public void userConsentUUID() {
+        assertEquals(consentUUIDMock, userConsent.uuid);
+    }
+
+    @Test
+    public void userConsentConsentString() throws JSONException {
+        assertEquals(userConsent.consentString, jsonConsentsMock.getString("euconsent"));
+    }
+
+    @Test
+    public void userConsentAcceptedCategories() throws JSONException {
+        ArrayList<String> acceptedCategories = userConsent.json2StrArr(jsonConsentsMock.getJSONArray("acceptedCategories"));
+        assertEquals(acceptedCategories, userConsent.acceptedCategories);
+    }
+
+    @Test
+    public void userConsentAcceptedVendors() throws JSONException {
+        ArrayList<String> acceptedVendors = userConsent.json2StrArr(jsonConsentsMock.getJSONArray("acceptedVendors"));
```

Comment:
Better make a JSONArray back from the ArrayList instead. This way you will compare two JSONArrays instead. You should not use the json2StrArr method as it is going to make your test unreliable. 
🤙 me and we can go over this in detail :)